server/api/api/endpoints/nuclio.py

```diff
@@ -546,12 +546,20 @@ def _deploy_nuclio_runtime(
             )
 
         if serving_to_monitor:
-            if monitoring_deployment.should_redeploy_monitoring_stream(
-                fn_image=fn.spec.image, client_version=client_version
+            if (
+                fn.spec.image.startswith("mlrun/")
+                and client_version
```

Comment:
just to my understanding, do we really need to check for a case where client_version was not provided? I would think we should fail this request fast, and later in the code assume we have it.